push.go

```diff
@@ -53,16 +53,53 @@ func (cmd *PushCommand) Run() error {
 type Sources []*Source
 
 type Source struct {
-	File          string      `yaml:"file,omitempty"`
-	ProjectID     string      `yaml:"project_id,omitempty"`
-	AccessToken   string      `yaml:"access_token,omitempty"`
-	FileFormat    string      `yaml:"file_format,omitempty"`
-	Params        *phraseapp.UploadParams `yaml:"params"`
+	File          string
+	ProjectID     string
+	AccessToken   string
+	FileFormat    string
+	Params        *phraseapp.UploadParams
 
 	RemoteLocales []*phraseapp.Locale
 	Extension     string
 }
 
+func (src *Source) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	m := map[string]interface{}{}
+	if err := unmarshal(m); err != nil {
+		return err
+	}
+
+	var ok bool
+	cfgErrStr := "configuration key %q has invalid value\nsee https://phraseapp.com/docs/developers/cli/configuration/"
+	for k, v := range m {
+		switch k {
+		case "file":
+			if src.File, ok = v.(string); !ok { return fmt.Errorf(cfgErrStr, k) }
+		case "project_id":
+			if src.ProjectID, ok = v.(string); !ok { return fmt.Errorf(cfgErrStr, k) }
+		case "access_token":
+			if src.AccessToken, ok = v.(string); !ok { return fmt.Errorf(cfgErrStr, k) }
+		case "file_format":
+			if src.FileFormat, ok = v.(string); !ok { return fmt.Errorf(cfgErrStr, k) }
+		case "params":
+			raw, ok := v.(map[interface{}]interface{});
+			if !ok { return fmt.Errorf(cfgErrStr, k) }
+
+			ps := map[string]interface{}{}
+			for k, v := range raw {
+				ps[k.(string)] = v
```

Comment:
in theory this would panic if params is e.g.
```
a: 1
b: 2
10: 11
```
as 10 would not be `"10"` but an integer.

I would say either use `fmt.Sprintf(k)` or check the cast and return an error if it is not a string(we probably do not have any non-string param keys anyway)?